Update example code to compile with Swift 5.0/5.1

GeekAndDad opened the pull request:
Update LazySequence documentation sample code for the "we can build a sequence that lazily computes the elements in the result of `scan`" example so that it will compile and work for Swift 5.0 & 5.1

GeekAndDad committed:
Update example code to compile with Swift 5.0/5.1

Took inspiration from Filter.swift as to modern syntax to use, though didn't convert `private` properties to `internal`.

theblixguy commented:
cc @airspeedswift @natecook1000

GeekAndDadDad commented:
Now that bugs.swift.org is back up aIssue has been created for this defect:  https://bugs.swift.org/browse/SR-11084

xwu commented on stdlib/public/core/LazySequence.swift:
```diff
@@ -68,6 +68,11 @@
+///         self.base = base
+///         self.nextPartialResult = nextPartialResult
+///       }
 ///       private var nextElement: ResultElement? // The next result of next().
```
Less boilerplate is better for example code, I think. It serves no didactic purpose here to make these members `private` in the example code, and it disables the synthesized initializer.

Instead of adding explicit initializers, let's delete these access modifiers to make the example compile again.

natecook1000 replied:
Agreed! I think you'll need to modify the initializer calls to include all the parameter labels, too. Thanks for catching this, @GeekAndDadDad!

GeekAndDadDad replied:
Can do.  That's how I initially made it work locally so I could move forward with my exploration of LazySequences.   Just thought it'd be better to expand it to match form of "real" code in Filter.swift.

Thanks for the feedback.

GeekAndDad committed:
simplify example code as requested

Removed private scope on properties to enable synthesized initializers; removed previously added explicit initializers; add necessary parameter label.

xwu commented on stdlib/public/core/LazySequence.swift:
```diff
@@ -68,25 +68,25 @@
 ///     }
 ///
-/// and finally, we can give all lazy sequences a lazy `scan` method:
+/// // and finally, we can give all lazy sequences a lazy `scan` method:
```
```suggestion
/// and finally, we can give all lazy sequences a lazy `scan` method:
```

This is correct as it was. (Code voice is automatically applied based on indenting.)

GeekAndDadDad replied:
Sorry, was trying to make it easier to copy and paste the code into a playground and have it run with fewer steps.  Thanks for catching this (!).

xwu commented on stdlib/public/core/LazySequence.swift:
```diff
@@ -68,25 +68,25 @@
 ///       func makeIterator() -> LazyScanIterator<Base.Iterator, ResultElement> {
 ///         return LazyScanIterator(
-///           nextElement: initial, base: base.makeIterator(), nextPartialResult)
+///           nextElement: initial, base: base.makeIterator(), nextPartialResult: nextPartialResult)
```
```suggestion
///           nextElement: initial, base: base.makeIterator(),
///           nextPartialResult: nextPartialResult)
```

GeekAndDad committed:
Update stdlib/public/core/LazySequence.swift

remove ill advised usability change

Co-Authored-By: Xiaodi Wu <xiaodi.wu@gmail.com>

GeekAndDad committed:
Update stdlib/public/core/LazySequence.swift

Co-Authored-By: Xiaodi Wu <xiaodi.wu@gmail.com>

xwu commented:
@swift-ci Please smoke test
